src/ring/middleware/json.clj

```diff
@@ -100,12 +102,19 @@
        (handler request respond raise)
        (respond malformed-response)))))
 
+(defn- json-stream
+  "Adapts the output stream from Cheshire to the input stream required by Ring."
+  [body options]
+  (piped-input-stream
+   (fn [out] (json/generate-stream body (writer out) options))))
+
 (defn json-response
   "Converts responses with a map or a vector for a body into a JSON response.
   See: wrap-json-response."
   [response options]
   (if (coll? (:body response))
-    (let [json-resp (update-in response [:body] json/generate-string options)]
+    (let [generator (if (:stream options) json-stream json/generate-string)
+          json-resp (update-in response [:body] generator options)]
```

Comment:
`:stream` keyword shouldn't be passed to generator.

Reply:
Laziness. Apologies.